diff --git "a/src/parseTestRecord.py" "b/src/parseTestRecord.py"
index 22d5664..5be3a5f 100644
--- "a/src/parseTestRecord.py"
+++ "b/src/parseTestRecord.py"
@@ -78,7 +78,7 @@ def yamlAttrParser(testRecord, attrs, name):
     parsed = yamlLoad(body)
 
     if (parsed is None):
-        print "Failed to parse yaml in name %s"%(name)
+        print ("Failed to parse yaml in name %s"%(name))
         return
 
     for key in parsed:
